packages/wrangler/src/sites.tsx

```diff
@@ -148,23 +146,55 @@ export async function syncAssets(
     console.log(`reading ${file}...`);
     const content = await readFile(file, "base64");
 
-    const assetKey = await hashAsset(hasher, file, content);
+    const assetKey = hashAsset(hasher, file, content);
     validateAssetKey(assetKey);
 
     // now put each of the files into kv
     if (!keys.has(assetKey)) {
       console.log(`uploading as ${assetKey}...`);
-      upload.push({
+      toUpload.push({
         key: assetKey,
         value: content,
         base64: true,
       });
     } else {
       console.log(`skipping - already uploaded`);
     }
+    // remove the key from the set so we know we've seen it
+    keys.delete(assetKey);
     manifest[path.relative(siteAssets.baseDirectory, file)] = assetKey;
   }
-  await putBulkKeyValue(accountId, namespace, upload, () => {});
+
+  // `keys` now contains the assets that we need to expire
+  const FIVE_MINUTES_LATER = Date.now() + FIVE_MINUTES;
+  let toExpire: KeyValue[] = [];
+  for (const asset of result) {
+    if (keys.has(asset.name)) {
+      console.log(`expiring unused ${asset.name}...`);
+      toExpire.push({
+        key: asset.name,
+        value: "", // we'll fill all the values in one go
+        expiration: asset.expiration || FIVE_MINUTES_LATER,
+        base64: true,
+      });
+    }
+  }
+
+  // TODO: batch these in groups if it causes problems
+  toExpire = await Promise.all(
+    toExpire.map(async (asset) => ({
+      ...asset,
+      // it would be great if we didn't have to do this fetch at all
+      value: await getKeyValue(accountId, namespace, asset.key),
```

Comment:
Is there a way to update a value's expiration time without having to fetch and send back the value? This is currently the most expensive part of this strategy.

Reply:
It would be ideal if the `bulk:kv` PUT API did not update values if the `value` property is missing.
Then we could do this whole update (additions and expirations) in one go.

Reply:
@koeninger

Reply:
@koeninger thoughts?

Reply:
@koeninger if this looks fine to you, and there's no additional APIs that can be provided, we'll land this and backport to wrangler 1.x.

Reply:
There's nothing different about bulk api in terms of efficiency, it's just a wrapper around multiple individual writes.  There's not a way to update expiration without a write.